File: backend/VibLink/Models/DTOs/Request/UserRegisterRequest.cs
using System.ComponentModel.DataAnnotations;

namespace VibLink.Models.DTOs.Request
{
    public record UserRegisterRequest
    {
        [Required]
        public string Email { get; init; } = string.Empty;

        [Required]
        public string Password { get; init; } = string.Empty;

        [Required]
        public string FirstName { get; init; } = string.Empty;

        [Required]
        public string LastName { get; init; } = string.Empty;

        public IFormFile? Picture { get; init; } = null;
    }
}

File: frontend/src/data/models/request/UserRegisterRequest.ts
export default class UserRegisterRequest {
  constructor(
    public readonly Email: string,
    public readonly Password: string,
    public readonly FirstName: string,
    public readonly LastName: string,
    public readonly Picture: File | null
  ) {}

  toJson(): object {
    return {
      email: this.Email,
      password: this.Password,
      firstName: this.FirstName,
      lastName: this.LastName
    }
  }

  toString(): string {
    return JSON.stringify(this.toJson())
  }
}
